### PR TITLE
Esp32 sparrow kit

### PR DESCRIPTION
**Pull request by robertalexa2000**
These patches add support for a new custom-made board, esp32-sparrow, which is built upon esp32-wrover-kit. The purpose to esp32-sparrow-kit is to enable by default support for various peripherals. So far, only LTR308 and SD cards are supported. The others will be addressed in separate future patches.

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by robertalexa2000** [comment]
@xiaoxiang781216 do you have any idea why the build failed?

File: boards/Kconfig [hunk 1]
```diff
@@ -253,6 +253,22 @@ config ARCH_BOARD_ESP32_LYRAT
...
+	select ARCH_HAVE_IRQBUTTONS if ESP32_GPIO_IRQ
+	---help---
+    ESP32-SPARROW-KIT is a custom made development board based on Espressif's
+    ESP32WROVER chip. It features the following integrated components:
+        ESP32-WROVER-B module
+        LTR308 ambient light sensor
+        BME180 sensor
+        MicroSD card slot
+        LCD screen
+        I2S microphone
```
Comment (pkarashchenko) [inline boards/Kconfig:270]: TABs

**Comment by pkarashchenko** [comment]
> @xiaoxiang781216 do you have any idea why the build failed? I can't see any errors in the log, am I missing something?

You based your config on some older version. Please normalize defconfig in this PR to reflect the actual Kconfig tree dependencies

**Comment by robertalexa2000** [comment]
> > @xiaoxiang781216 do you have any idea why the build failed? I can't see any errors in the log, am I missing something?
> 
> You based your config on some older version. Please normalize defconfig in this PR to reflect the actual Kconfig tree dependencies

Could you please be a little more explicit? What do you mean by the actaul Kconfig dependencies? The defconfig is based on the esp32-wrover-kit nsh and mmcsd defconfigs so I thought it would be fine.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by hartmannathan** [comment]
> No board-level README.txt file to describe the board?

One useful thing to document: since this board support is being made public, how can someone obtain one of these boards? (Is it manufactured by a company? Is there a public reference design? Etc.)

**Comment by robertalexa2000** [comment]
> > No board-level README.txt file to describe the board?
> 
> One useful thing to document: since this board support is being made public, how can someone obtain one of these boards? (Is it manufactured by a company? Is there a public reference design? Etc.)

I could add a README but I don't know what any extra info it can provide, besides what I summarized in boards/Kconfig.

Sparrow-kit is basically an ESP32-Wrover with a few extra peripherals soldered into it. It's not manufactured by a company, it was custom-made for a university course. Its purpose is to allow the user to enable all these add-ons by default, without the need of manually checking the necessary configs.

Maybe instead of a README, I could add these details in boards/Kconfig?

**Comment by hartmannathan** [comment]
> Maybe instead of a README, I could add these details in boards/Kconfig?

Sure, that would be good enough. The explanation you wrote above is perfectly fine. Thanks!

**Review by hartmannathan — APPROVED** [review]

File: boards/xtensa/esp32/common/src/esp32_ltr308.c [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  struct i2c_master_s *i2c;
+  int ret;
+
+  sninfo("Initializing LTR308!\n");
+
+  /* Initialize LTR308 */
+
+  i2c = esp32_i2cbus_initialize(busno);
+
+  if (i2c)
```
Comment (pkarashchenko) [inline boards/xtensa/esp32/common/src/esp32_ltr308.c:69]:
```suggestion
  i2c = esp32_i2cbus_initialize(busno);
  if (i2c !- NULL)
```
optional

**Review by pkarashchenko — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**